### PR TITLE
Allowing users to provide additional http headers for Prometheus Client 

### LINKED ISSUE #21394: Allow users to provide additional headers  as properties to connect to prometheus endpoint
Prometheus endpoint can require additional headers, for example, X-Origin> 
Allow users to pass headers and their values as properties

### PR DESCRIPTION
**Pull request by amitkharb**
## Description

Prometheus endpoint may require additional http headers. Adding the configuration properties for the same
Example of additional header  "X-Scope-orgID" , "default"

Fixes #21394

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Prometheus
* TBD. ({issue}`21394`)
```

### CODE REVIEW

File: plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java [hunk 1]
```diff
@@ -213,13 +213,19 @@ private static Interceptor basicAuth(String user, String password)
                 .build());
     }
 
-    private static Interceptor tokenAuth(String accessToken, String httpAuthHeaderName)
+    private static Interceptor tokenAuth(String accessToken, String httpAuthHeaderName, Map<String, String> additionalHeaders)
```
Comment (ebyhr) [inline plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java:216]: Why not set additional HTTP headers in basic auth? Can we revert this `tokenAuth` method change and set those headers regardless of authentication?

File: plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java [hunk 2]
```diff
@@ -41,6 +43,7 @@ public class PrometheusConnectorConfig
     private File bearerTokenFile;
     private String user;
     private String password;
+    private final Map<String, String> additionalHeaders = new HashMap<>();
```
Comment (ebyhr) [inline plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java:46]:
```suggestion
    private final Map<String, String> additionalHeaders = ImmutableMap.of();
```

File: plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java [hunk 3]
```diff
@@ -152,6 +155,28 @@ public PrometheusConnectorConfig setPassword(String password)
...
     }
 
+    public Map<String, String> getAdditionalHeaders()
+    {
+        return this.additionalHeaders;
+    }
+
+    @Config("prometheus.http.additional.headers")
+    @ConfigDescription("Additional headers to be sent with the HTTP request to Prometheus")
+    public PrometheusConnectorConfig setAdditionalHeaders(String additionalHeaders)
```
Comment (ebyhr) [inline plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java:165]: You can refer to `ThriftHttpMetastoreConfig#setAdditionalHeaders` method when converting string to map.

File: plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java [hunk 4]
```diff
@@ -152,6 +155,28 @@ public PrometheusConnectorConfig setPassword(String password)
...
+            for (String headerPair : headerPairs) {
+                String[] headerParts = headerPair.split("=");
+                if (headerParts.length != 2) {
+                    throw new IllegalArgumentException("Invalid header format: " + headerPair);
+                }
+                this.additionalHeaders.putIfAbsent(headerParts[0], headerParts[1]);
+            }
+        }
+        return this;
+    }
```
Comment (ebyhr) [inline plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java:178]: Can we add a validation that denies `httpAuthHeaderName` in additional headers?

**Review by ebyhr — COMMENTED** [review]
Please squash commits into one.

**Comment by amitkharb** [comment]
I have messed up this branch, I will create another PR, closing this one

**Comment by amitkharb** [comment]
Thank you @ebyhr , having a quick look at it. 
Somehow I messed up my branch and end up creating a new PR, 
All your comments are addressed in https://github.com/trinodb/trino/pull/22300/files
Sorry for inconvenience

**Pull request closed without merging**